subcommittee/coding-guidelines/initiatives/safe-use-of-unsafe-guidelines/safe-use-of-unsafe-guidelines-proposal.md

```diff
@@ -0,0 +1,56 @@
+# Safe Use of Unsafe Guidelines Proposal
+
+## Motivation
+
+By default, Rust is a safe programming language. Outside of  `unsafe`,  Rust programs do not have undefined behavior or invalid memory access.
+
+There are reasonable use cases to use the `unsafe` keyword such as:
+
+* interop through Rust’s C FFI  
+* complex safety invariants that cannot be verified by the Rust compiler
```

Comment:
- accessing memory mapped I/O registers.

Reply:
I like adding this. I'm in favor of merging as-is without it and then for folks with other thoughts opening an issue and PR. Great way for folks to be able to have control over the changes. (such as yourself :v:)

WDYT?

Reply:
Followed up with Jonas on Zulip, going to mark this as resolved for now and merge.